redwood-treehouse-lazylayout-view/src/main/kotlin/app/cash/redwood/treehouse/lazylayout/view/ViewLazyColumn.kt

```diff
@@ -62,64 +54,141 @@ internal class ViewLazyColumn<A : AppService>(
       layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
     }
     value.adapter = adapter
+    value.recycledViewPool.setMaxRecycledViews(adapter.getItemViewType(RecyclerView.NO_POSITION), 100)
   }
 
   override fun intervals(intervals: List<LazyListIntervalContent>) {
-    adapter.submitList(
-      intervals.flatMap { interval ->
-        List(interval.count) { index ->
-          LazyContentItem(index, interval.itemProvider)
-        }
-      },
-    )
+    adapter.submitList(intervals.flatMap { interval -> List(interval.count) { interval } })
   }
 
-  private class LazyContentItemListAdapter<A : AppService>(
+  private class LazyListIntervalContentAdapter<A : AppService>(
     private val treehouseApp: TreehouseApp<A>,
     private val widgetSystem: TreehouseView.WidgetSystem<A>,
-    private val contentHeight: Int,
-  ) : ListAdapter<LazyContentItem, ViewHolder<A>>(LazyContentItemDiffCallback) {
+  ) : ListAdapter<LazyListIntervalContent, LazyListIntervalContentAdapter.ViewHolder<A>>(LazyListIntervalContentDiffCallback) {
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<A> {
       val container = FrameLayout(parent.context).apply {
-        layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, contentHeight)
+        layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
       }
       return ViewHolder(container, treehouseApp, widgetSystem)
     }
 
     override fun onBindViewHolder(holder: ViewHolder<A>, position: Int) {
-      val itemContent = currentList[position]
-      holder.treehouseWidgetView.setContent {
-        itemContent.item.get(itemContent.index)
-      }
+      holder.itemTreehouseWidgetViewLoadState = LoadState.Loading
+      holder.interval = getItem(position)!!
```

Comment:
Is the variable an _interval_ or is it an _item_ ? Since the position is already applied I don’t think interval makes sense as a name?